lib/src/front_end/piece_factory.dart

```diff
@@ -573,6 +573,29 @@ mixin PieceFactory {
     return builder.build();
   }
 
+  /// Creates a [Piece] for an index expression whose [target] has already been
+  /// converted to a piece.
+  ///
+  /// The [target] may be `null` if [index] is an index expression for a
+  /// cascade section.
+  Piece createIndexExpression(Piece? target, IndexExpression index) {
+    // TODO(tall): Consider whether we should allow splitting between
```

Comment:
Just my two cents, but I think we should. At least reading the first postfix test, I find myself expecting a split.

If this doesn't happen, is there just no split at all for all chained index expressions?

Reply:
Yeah, if you have a long series of index operations, it is nice to be able to split. But, outside of weird contrived formatter tests, they're pretty rare. I think we probably will end up supporting in the new style (the old one does), but it was kind of tricky so I figured I would leave a TODO and kick it down the road a bit.

>  is there just no split at all for all chained index expressions?

Correct, it just keeps on going on one line.